Exercises/chapter6/e3-iterable-groups.js

```diff
@@ -0,0 +1,75 @@
+class Group {
+  constructor() {
+    this.values = [];
+  }
+
+  static from(array) {
+    let group = new Group();
+    array.forEach(e => group.add(e));
+    return group;
+  }
+
+  add(value) {
+    if (!this.has(value)) {
+      this.values.push(value);
+    }
+  }
+
+  delete(value) {
+    let indexToDelete = this.values.indexOf(value);
+    if (indexToDelete !== -1) {
+      this.values.splice(indexToDelete, 1);
+    }
+  }
+
+  has(value) {
+    return this.values.indexOf(value) !== -1;
+  }
+}
+
+class IterableGroup extends Group {
```

Comment:
I think you'd be better off just adding the iterator method to the group class, but some (wrong) people would agree that using inheritance like this is better.